src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -862,8 +894,11 @@ class DocBaseViewer extends BaseViewer {
 
         this.loadUI();
 
-        // Set current page to previously opened page or first page
-        this.setPage(this.getCachedPage());
+        const { pagesCount, currentScale } = this.pdfViewer;
+
+        // Set page to the user-defined page, previously opened page, or first page
+        const startPage = this.startPageNum || this.getCachedPage();
```

Comment:
You're not using this.startPageNum anywhere else, why not just set it here locally instead of a class variable?

`const startPage = this.getStartPage() || this.getCachedPage();`

Edit: Just saw Tony's comment, that makes sense. Would it make sense to standardize on the name across viewers?